cmd/jaeger/internal/integration/integration.go

```diff
@@ -4,129 +4,61 @@
-	require.True(t, ok, "no jaeger storage exporter found in the config")
-
-	telemetrySettings := componenttest.NewNopTelemetrySettings()
-	telemetrySettings.Logger = tel.Logger()
-
-	receiver := datareceivers.NewJaegerStorageDataReceiver(
-		telemetrySettings,
-		exporterCfg.TraceStorage,
-		storageCfg,
-	)
-	return receiver
+	binaryProcess *os.Process
 }
 
-func (s *StorageIntegration) Test(t *testing.T) {
-	if os.Getenv("STORAGE") != s.Name {
-		t.Skipf("Integration test against Jaeger-V2 %[1]s skipped; set STORAGE env var to %[1]s to run this", s.Name)
+// e2eInitialize starts the Jaeger-v2 collector with the provided config file,
+// it also initialize the SpanWriter and SpanReader below.
+func (s *E2EStorageIntegration) e2eInitialize(t *testing.T) {
+	cmd := exec.Cmd{
+		Path: "./cmd/jaeger/jaeger",
+		Args: []string{"jaeger", "--config", s.ConfigFile},
+		// Change the working directory to the root of this project
+		// since the binary config file jaeger_query's ui_config points to
+		// "./cmd/jaeger/config-ui.json"
+		Dir:    "../../../..",
+		Stdout: os.Stderr,
+		Stderr: os.Stderr,
 	}
+	require.NoError(t, cmd.Start())
+	s.binaryProcess = cmd.Process
 
-	factories, err := internal.Components()
-	require.NoError(t, err)
-
-	dataProvider := testbed.NewGoldenDataProvider(
-		"fixtures/generated_pict_pairs_traces.txt",
-		"fixtures/generated_pict_pairs_spans.txt",
-		"",
-	)
-	sender := testbed.NewOTLPTraceDataSender(testbed.DefaultHost, 4317)
-	receiver := s.newDataReceiver(t, factories)
+	spanWriter := createSpanWriter(otlpPort)
+	require.NoError(t, spanWriter.Start())
+	s.SpanWriter = spanWriter
 
-	runner := testbed.NewInProcessCollector(factories)
-	validator := testbed.NewCorrectTestValidator(
-		sender.ProtocolName(),
-		receiver.ProtocolName(),
-		dataProvider,
-	)
-	correctnessResults := &testbed.CorrectnessResults{}
-
-	config, err := os.ReadFile(s.ConfigFile)
-	require.NoError(t, err)
-	t.Log(string(config))
-
-	configCleanup, err := runner.PrepareConfig(string(config))
-	require.NoError(t, err)
-	defer configCleanup()
-
-	tc := testbed.NewTestCase(
-		t,
-		dataProvider,
-		sender,
-		receiver,
-		runner,
-		validator,
-		correctnessResults,
-	)
-	defer tc.Stop()
-
-	tc.EnableRecording()
-	tc.StartBackend()
-	tc.StartAgent()
-
-	tc.StartLoad(testbed.LoadOptions{
-		DataItemsPerSecond: 1024,
-		ItemsPerBatch:      1,
-	})
-	tc.Sleep(5 * time.Second)
-	tc.StopLoad()
-
-	tc.WaitForN(func() bool {
-		return tc.LoadGenerator.DataItemsSent() == tc.MockBackend.DataItemsReceived()
-	}, 10*time.Second, "all data items received")
+	spanReader := createSpanReader(ports.QueryGRPC)
+	require.NoError(t, spanReader.Start())
+	s.SpanReader = spanReader
+}
 
-	tc.StopBackend()
+func (s *E2EStorageIntegration) e2eCleanUp(t *testing.T) {
+	require.NoError(t, s.SpanReader.(io.Closer).Close())
+	require.NoError(t, s.SpanWriter.(io.Closer).Close())
 
-	tc.ValidateData()
+	s.binaryProcess.Kill()
```

Comment:
nit: this should probably be in `defer` inside e2eCleanUp, to ensure that it's always run (the two require above could throw and exit). I think even better it to add it to t.Cleanup in e2eInitialize